flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/utils/ParquetRecordReader.java

```diff
@@ -0,0 +1,196 @@
+	private RecordReader<T> recordReader;
+	private boolean strictTypeChecking = true;
+	private long totalCountLoadedSoFar = 0;
+
+	public ParquetRecordReader(ReadSupport<T> readSupport, MessageType readSchema, Filter filter) {
+		this.readSupport = readSupport;
+		this.readSchema = readSchema;
+		this.filter = checkNotNull(filter, "filter");
+	}
+
+	public ParquetRecordReader(ReadSupport<T> readSupport, MessageType readSchema) {
+		this(readSupport, readSchema, FilterCompat.NOOP);
+	}
+
+	public void initialize(ParquetFileReader reader, Configuration configuration) {
+		this.reader = reader;
+		FileMetaData parquetFileMetadata = reader.getFooter().getFileMetaData();
+		this.fileSchema = parquetFileMetadata.getSchema();
+		Map<String, String> fileMetadata = parquetFileMetadata.getKeyValueMetaData();
+		ReadSupport.ReadContext readContext = readSupport.init(new InitContext(
+			configuration, toSetMultiMap(fileMetadata), readSchema));
+
+		this.columnIOFactory = new ColumnIOFactory(parquetFileMetadata.getCreatedBy());
+		this.readSchema = readContext.getRequestedSchema();
+		this.recordMaterializer = readSupport.prepareForRead(
+			configuration, fileMetadata, readSchema, readContext);
+		this.total = reader.getRecordCount();
+		reader.setRequestedSchema(readSchema);
+	}
+
+	private void checkRead() throws IOException {
+		if (current == totalCountLoadedSoFar) {
+			PageReadStore pages = reader.readNextRowGroup();
+			recordReader = createRecordReader(pages);
+			totalCountLoadedSoFar += pages.getRowCount();
+			currentBlock++;
+		}
+	}
+
+	public void close() throws  IOException {
+		if (reader != null) {
+			reader.close();
+		}
+	}
+
+	@CheckReturnValue(when = When.NEVER)
+	public T nextRecord() {
+		return currentValue;
+	}
+
+	public void seek(long syncedBlock) throws IOException {
+		PageReadStore pages = null;
+		while (syncedBlock > 0) {
+			pages = reader.readNextRowGroup();
+		}
+
+		recordReader = createRecordReader(pages);
+	}
+
+	private RecordReader<T> createRecordReader(PageReadStore pages) throws IOException {
+		if (pages == null) {
+			throw new IOException("Expecting more rows but reached last block. Read " + current + " out of " + total);
+		}
+		MessageColumnIO columnIO = columnIOFactory.getColumnIO(readSchema, fileSchema, strictTypeChecking);
+		// TODO (hpeter) enable filter later
+		RecordReader<T> recordReader = columnIO.getRecordReader(pages, recordMaterializer, FilterCompat.NOOP);
+		return recordReader;
+	}
+
+	public long getCurrentBlock() {
+		return currentBlock;
+	}
+
+	public boolean reachEnd() {
+		return current >= total;
+	}
+
+	public boolean hasNextRecord() throws IOException {
+		boolean recordFound = false;
+
+		while (!recordFound) {
+			// no more records left
+			if (current >= total) {
+				return false;
+			}
+
+			try {
+				checkRead();
+				current++;
+				try {
+					currentValue = recordReader.read();
+				} catch (RecordMaterializationException e) {
+					LOG.debug("skipping a corrupt record");
```

Comment:
Add more detail to the log like file name, block number, record number and increase log level to ERROR as you did below.

Depending on the configuration, we might even want to throw an exception and cancel the job.